compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesBindingGeneratorTest.kt

```diff
@@ -304,4 +308,125 @@ class ContributesBindingGeneratorTest {
       )
     }
   }
+
+  @Test fun `there are multiple hints for multiple contributed bindings`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+
+      interface ParentInterface
+
+      @ContributesBinding(Any::class)
+      @ContributesBinding(Unit::class)
+      class ContributingInterface : ParentInterface
+      """
+    ) {
+      assertThat(contributingInterface.hintBinding?.java).isEqualTo(contributingInterface)
+      assertThat(contributingInterface.hintBindingScopes).containsExactly(Any::class, Unit::class)
+    }
+  }
+
+  @Test fun `the scopes for multiple contributions have a stable sort`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+
+      interface ParentInterface
+
+      @ContributesBinding(Any::class)
+      @ContributesBinding(Unit::class)
+      class ContributingInterface : ParentInterface
+      
+      @ContributesBinding(Unit::class)
+      @ContributesBinding(Any::class)
+      class SecondContributingInterface : ParentInterface
+      """
+    ) {
+      assertThat(contributingInterface.hintBindingScopes)
+        .containsExactly(Any::class, Unit::class)
+        .inOrder()
+
+      assertThat(secondContributingInterface.hintBindingScopes)
+        .containsExactly(Any::class, Unit::class)
+        .inOrder()
+    }
+  }
+
+  @Test fun `there are multiple hints for contributed bindings with fully qualified names`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+      
+      interface ParentInterface
+
+      @ContributesBinding(Any::class)
+      @com.squareup.anvil.annotations.ContributesBinding(Unit::class)
+      class ContributingInterface : ParentInterface
+      """
+    ) {
+      assertThat(contributingInterface.hintBinding?.java).isEqualTo(contributingInterface)
+      assertThat(contributingInterface.hintBindingScopes).containsExactly(Any::class, Unit::class)
+    }
+  }
+
+  @Test fun `multiple contribution can have different bound types`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+      
+      interface ParentInterface1
+      interface ParentInterface2
+
+      @ContributesBinding(Any::class, boundType = ParentInterface1::class)
+      @ContributesBinding(Unit::class, boundType = ParentInterface2::class)
+      class ContributingInterface : ParentInterface1, ParentInterface2
+      """
+    ) {
+      assertThat(contributingInterface.hintBinding?.java).isEqualTo(contributingInterface)
+      assertThat(contributingInterface.hintBindingScopes).containsExactly(Any::class, Unit::class)
```

Comment:
I feel like we need an additional (or different) assert here to verify the different boundTypes were maintained. It would be possible for this test to continue passing even if the compile source was changed to not specify any bound types.

Reply:
Hm, this test only generates the hint and not the actual binding method. I agree that it's confusing and moved the test into the `BindingModuleGeneratorTest` file.